src/lesson02/include/arm/sysregs.h

```diff
@@ -6,14 +6,14 @@
 // ***************************************
 
 #define SCTLR_RESERVED                  (3 << 28) | (3 << 22) | (1 << 20) | (1 << 11)
-#define SCTLR_EE_LITTLE_ENDIAN          (0 << 25)
-#define SCTLR_EOE_LITTLE_ENDIAN         (0 << 24)
-#define SCTLR_I_CACHE_DISABLED          (0 << 12)
-#define SCTLR_D_CACHE_DISABLED          (0 << 2)
-#define SCTLR_MMU_DISABLED              (0 << 0)
+#define SCTLR_EE_LITTLE_ENDIAN          ~(1 << 25)
+#define SCTLR_EOE_LITTLE_ENDIAN         ~(1 << 24)
+#define SCTLR_I_CACHE_DISABLED          ~(1 << 12)
+#define SCTLR_D_CACHE_DISABLED          ~(1 << 2)
+#define SCTLR_MMU_DISABLED              ~(1 << 0)
 #define SCTLR_MMU_ENABLED               (1 << 0)
 
-#define SCTLR_VALUE_MMU_DISABLED	(SCTLR_RESERVED | SCTLR_EE_LITTLE_ENDIAN | SCTLR_I_CACHE_DISABLED | SCTLR_D_CACHE_DISABLED | SCTLR_MMU_DISABLED)
+#define SCTLR_VALUE_MMU_DISABLED	(SCTLR_RESERVED & SCTLR_EE_LITTLE_ENDIAN & SCTLR_I_CACHE_DISABLED & SCTLR_D_CACHE_DISABLED & SCTLR_MMU_DISABLED)
```

Comment:
I think the resulting value of SCTLR_VALUE_MMU_DISABLED is going to be qual to the SCTLR_RESERVED, right? This was the case before this fix as well. What is the goal of this fix?

Reply:
As the document wrote, you are going to clear the specific bits to enable/disable some behaviors, but the current version is bitwise or SCTLR_RESERVED with a few 0, and SCTLR_VALUE_MMU_DISABLED will always equal to SCTLR_RESERVED isn't it? 
I think the logic is we have a value SCTLR_RESERVED and we are going to clear some bits in it, so we make some bit mask via ~(1<<offset) and bit wise and them together. Or why not just assign SCTLR_RESERVED to SCTLR_VALUE_MMU_DISABLED without bit wise or with a few 0.

Reply:
No, we are not going to clear some bits on SCTLR_RESERVED. Those bits are set based on the specification and should always be 1. The whole purpose of the other variables is to provide users an easy way to change some common settings, if they want to do so. For example, if you want to disable i-cache you set SCTLR_I_CACHE_DISABLED to 1 and recompile.

Reply:
Ok I think I understand your point, just some misunderstood about these macros.
So should I close this pull request, or I should do something else?
Thank you!

Reply:
You can revert changes in sysregs.h and then I'll be able to merge your PR

Reply:
By my mistake I deleted the file, I think I close this pr first and I will raise another one later, after I complete more exercises
Thank you!